azurerm/resource_arm_sql_managed_instance.go

```diff
@@ -0,0 +1,236 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/sql/mgmt/2015-05-01-preview/sql"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmSqlMiServer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmSqlMiServerCreateUpdate,
+		Read:   resourceArmSqlMiServerRead,
+		Update: resourceArmSqlMiServerCreateUpdate,
+		Delete: resourceArmSqlMiServerDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateMsSqlServerName,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"sku": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"GP_Gen4",
+								"GP_Gen5",
+							}, true),
+							DiffSuppressFunc: suppress.CaseDifference,
+						},
+
+						"capacity": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+
+						"tier": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"family": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"administrator_login": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"administrator_login_password": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+
+			"vcores": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  8,
+				ValidateFunc: validate.IntInSlice([]int{
+					8,
```

Comment:
A few weeks ago a smaller number of vcores (4) was introduced:
https://docs.microsoft.com/en-us/azure/sql-database/sql-database-managed-instance-resource-limits#instance-level-resource-limits

It seems worthwhile to have this new resource up to date.